### PR TITLE
Stop Official CI pipeline trigger on private branch

### PR DESCRIPTION
**Pull request by erdembayar**
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/1468

Regression? Last working version:

## Description
It's [recommended change](https://developercommunity.visualstudio.com/t/yaml-trigger-starting-builds-on-branches/1681956?space=21&entry=myfeedback) by Azdo Team to stop private branch trigger on Official CI pipeline. 

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [x] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

File: eng/pipelines/official.yml [hunk 1]
```diff
@@ -1,13 +1,11 @@
 pr: none
 trigger:
-  batch: false
-  branches:
+ branches:
     include:
-      - dev
-      - release-*-MSRC
+      - main
```
Comment (nkolev92) [inline eng/pipelines/official.yml]: I don't think you should be blindly following their recommendation. 
We use dev, not main.

**Review by nkolev92 — CHANGES_REQUESTED** [review]

File: eng/pipelines/official.yml (hunk 1 above)
Comment (erdembayar) [reply eng/pipelines/official.yml]: Sure, pushed changes. Please check now.

File: eng/pipelines/official.yml [hunk 2]
```diff
@@ -4,10 +4,9 @@ trigger:
   branches:
     include:
       - dev
-      - release-*-MSRC
+      - release/release-*-MSRC
```
Comment (nkolev92) [inline eng/pipelines/official.yml:7]: Why do we need the `release/` prefix?

The docs here don't really cover that: https://docs.microsoft.com/en-us/azure/devops/pipelines/repos/azure-repos-git?view=azure-devops&tabs=yaml#ci-triggers

Comment (zivkan) [reply eng/pipelines/official.yml:7]: personally, I don't think this will work. Our release branches are `release-z.y.x`, not `release/release-z.y.x`.

Comment (erdembayar) [reply eng/pipelines/official.yml:7]: @zivkan 
Could you ask from Azdo team about if we need `release/` prefix to make things work? Somehow, I can't sign in and comment under issue you [created](https://developercommunity.visualstudio.com/t/yaml-trigger-starting-builds-on-branches/1681956?space=21&entry=myfeedback).

Comment (zivkan) [reply eng/pipelines/official.yml:7]: I've been in non-stop meetings all day today, so I haven't had time to yet. As I said in one of those meetings, I am going to.

**Comment by erdembayar** [comment]
Closing since @zivkan already working on other solution.

**Pull request closed without merging**